edweiss-app/app/(app)/lectures/slides/index.tsx

```diff
@@ -45,36 +45,42 @@ const LectureScreen: ApplicationRoute = () => {
     useListenToMessages((msg) => {
         if (msg.type == "go_to_slide") {
             setPage(msg.slideIndex);
+            setCurrentPage(msg.slideIndex); // Sync current page number display 
         }
     });
 
     // Hooks
     const [numPages, setNumPages] = useState<number>(0);        // Total number of pages in the PDF
-    const [page, setPage] = useState<number>(0);                // Current page, starting from 1
+    const [page, setPage] = useState<number>(1);                // Current page, starting from 1
+    const [currentPage, setCurrentPage] = useState<number>(1);  // Track swiped or active page
     const [uri, setUri] = useState<string>('');                 // Url state
     // UI display setting's hooks
-    const [isLandscape, setIsLandscape] = useState<boolean>(false);       // Landscape display boolean for different UI
+    const [isLandscape, setIsLandscape] = useState<boolean>(true);       // Landscape display boolean for different UI
     const [isFullscreen, setIsFullscreen] = useState<boolean>(false);    // FullScreen display of pdf toggle
-    const [widthPercent, setWidthPercent] = useState<string[]>([]);
-    const [heightPercent, setHeightPercent] = useState<string[]>([]);
-    const [question, setQuestion] = useState<string>('');
+
 
     const [lectureDoc] = usePrefetchedDynamicDoc(CollectionOf<Lecture>(`courses/${courseName}/lectures`), lectureId, undefined);
     const questionsDoc = useDynamicDocs(CollectionOf<Question>(`courses/${courseName}/lectures/${lectureId}/questions`));
 
     // Load images when the component mounts
     useEffect(() => {
-        if (lectureDoc) {
+        if (lectureDoc?.data.pdfUri) {
             getUri();
         }
     }, [lectureDoc]);
 
+
+    // Landscape display for the screen
+    const setLandscape = async () => {
+        await ScreenOrientation.lockAsync(ScreenOrientation.OrientationLock.LANDSCAPE);
+    };
```

Comment:
I like this, I think as you have become the master in rotation update, could you consider updating the wiki so that we can all leverage this feature cool feature.

Reply:
Yes I also like it very much, but there could still be some issues if used incorrectly that I would like to keep searching for a better solution so that everyone can use it without much worry. 
Since router.back() can be called anytime there is hardly any way to "unlock" this rotation before changing screen, which could end up creating some issues where a screen that is not well-prepared for landscape mode display would get stuck in that mode and would be unable to rotate it. 
However if I do not find this solution, I would then have to ask that every single screen contains a 
   ```
 useEffect(() => {
          ScreenOrientation.unlockAsync();
    }, []);
```
that would unlock any previously locked orientation to avoid issues.

I will update everyone once this is more well defined

Reply:
Seems totally reasonable, you are the expert.